Remove "bgpropertiesAttr"

Ahmad-S792 opened the pull request:
#### a6647584c85563e818f37e332d9f68d6bd2c3ae4
<pre>
Remove &quot;bgpropertiesAttr&quot;

Remove &quot;bgpropertiesAttr&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=249888">https://bugs.webkit.org/show_bug.cgi?id=249888</a>

Reviewed by Tim Nguyen.

This patch is to align WebKit with Blink / Chromium, Gecko / Firefox and Web-Specifications.

Inspired - <a href="https://chromium.googlesource.com/chromium/blink/+/278d9993a71dfaa8f8e324ac021313ea038e5e44">https://chromium.googlesource.com/chromium/blink/+/278d9993a71dfaa8f8e324ac021313ea038e5e44</a>

This patch is to remove &apos;bgproperties&quot;, which was supported by Internet Explorer but not by Gecko
and never standardized.
Blink also removed it in 2014 and now only WebKit / Safari support it and this patch is to remove last
usage and align all browser engines.

* Source/WebCore/html/HTMLAttributeNames.in: Remove &apos;bgproperties&apos;
* Source/WebCore/html/HTMLBodyElement.cpp:
(HTMLBodyElement::hasPresentationalHintsForAttribute): Remove &apos;bgpropertiesAttr&apos;
(HTMLBodyElement::collectPresentationalHintsForAttribute): Ditto

Canonical link: <a href="https://commits.webkit.org/258704@main">https://commits.webkit.org/258704@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0e41269ebc809916527bf007399b5a082985fa47

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102720 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11840 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35746 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111976 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172216 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106686 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12852 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2748 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94975 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109665 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108496 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9859 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93071 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37508 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24583 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79251 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5298 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26001 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5447 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2451 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11467 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45492 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5980 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7189 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/html/HTMLBodyElement.cpp:
```diff
@@ -3,7 +3,8 @@
  *           (C) 2001 Dirk Mueller (mueller@kde.org)
- * Copyright (C) 2004-2021 Apple Inc. All rights reserved.
+ * Copyright (C) 2004-2023 Apple Inc. All rights reserved.
+ * Copyright (C) 2014 Google Inc. All rights reserved.
```
This is a code removal, so not sure there's any Google IP involved here.

Ahmad-S792 replied:
I was just merging Google patch and using it as reference, hence I added it. If you want, I can get rid of it.

Ahmad-S792 replied:
Technically, you are right, it is not merge but just removal and we are just using it as inspiration to follow. Noted, I am pushing updated commit to remove this line and also improve commit message to change "Merge" to "Inspired".

Ahmad-S792 committed:
Remove "bgpropertiesAttr"

Remove "bgpropertiesAttr"
https://bugs.webkit.org/show_bug.cgi?id=249888

Reviewed by Tim Nguyen.

This patch is to align WebKit with Blink / Chromium, Gecko / Firefox and Web-Specifications.

Inspired - https://chromium.googlesource.com/chromium/blink/+/278d9993a71dfaa8f8e324ac021313ea038e5e44

This patch is to remove 'bgproperties", which was supported by Internet Explorer but not by Gecko
and never standardized.
Blink also removed it in 2014 and now only WebKit / Safari support it and this patch is to remove last
usage and align all browser engines.

* Source/WebCore/html/HTMLAttributeNames.in: Remove 'bgproperties'
* Source/WebCore/html/HTMLBodyElement.cpp:
(HTMLBodyElement::hasPresentationalHintsForAttribute): Remove 'bgpropertiesAttr'
(HTMLBodyElement::collectPresentationalHintsForAttribute): Ditto

Canonical link: https://commits.webkit.org/258704@main
